frontend/src/components/organisms/ManageAttendees.tsx

```diff
@@ -125,6 +134,212 @@ const AttendeesTable = ({
   );
 };
 
+interface modalProps {
+  handleClose: () => void;
+  eventDetails?: FormValues;
+}
+
+type FormValues = {
+  startDate: string;
+  endDate: string;
+  startTime: string;
+  endTime: string;
+};
+
+/** A duplicate event modal body */
+const ModalBody = ({ handleClose, eventDetails }: modalProps) => {
+  const { user } = useAuth();
+  const queryClient = useQueryClient();
+
+  /** State variables for the notification popups */
+  const [successNotificationOpen, setSuccessNotificationOpen] = useState(false);
+  const [errorNotificationOpen, setErrorNotificationOpen] = useState(false);
+
+  /** React hook form */
+  const {
+    register,
+    handleSubmit,
+    watch,
+    getValues,
+    setValue,
+    control,
+    formState: { errors },
+  } = useForm<FormValues>(
+    eventDetails
+      ? {
+          defaultValues: {
+            startDate: eventDetails.startDate,
+            endDate: eventDetails.endDate,
+            startTime: eventDetails.startTime,
+            endTime: eventDetails.endTime,
+          },
+        }
+      : {}
+  );
+
+  /** Handles form errors for time and date validation */
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const timeAndDateValidation = () => {
+    const { startTime, startDate, endTime, endDate } = getValues();
+    const startDateTime = convertToISO(startTime, startDate);
+    const endDateTime = convertToISO(endTime, endDate);
+    if (new Date(startDateTime) >= new Date(endDateTime)) {
+      setErrorNotificationOpen(true);
+      setErrorMessage(
+        "End Date and Time must be later than Start Date and Time"
+      );
+      return false;
+    } else {
+      setErrorMessage(null);
+    }
+    return true;
+  };
+
+  const back = () => {
+    router.push("/events/view");
+  };
+
+  /** Tanstack mutation for creating a new event */
+  const { mutateAsync, isPending, isError, isSuccess } = useMutation({
+    mutationFn: async (formData: FormValues) => {
+      const eventid = router.query.eventid as string;
```

Comment:
ModalBody is a child component, the eventID is already available in the parent component, so there is no need to make another request in the child component. You can simply pass it as a prop

Reply:
I think the userID is also available in the parent component too

Reply:
fixed; I couldn't find userid anywhere in the parent component so I just kept the query in the child component